client/src/lib/VolunteerProvider.js

```diff
@@ -0,0 +1,134 @@
+import React, { useState, createContext, useContext } from "react";
+import { Route, Redirect } from 'react-router-dom';
+
+const VolunteerContext = createContext(null);
+
+function VolunteerProvider({ children }) {
+  // Use localstorage for the moment until we get Sessions figured out.
+  const storedProfile = localStorage.getItem('volunteer');
+  let defaultProfile = {};
+  if (storedProfile) { // If profile is in localstorage, use that.
+    defaultProfile = JSON.parse(storedProfile);
+  } else {
+    defaultProfile = {
+      isAuthenticated: false,
+      notRegistered: false,
+      email: '',
+      registrationStep: 0,
+    };
+  }
+  
+  const [profile, setProfile] = useState(defaultProfile);
+
+  /**
+   * Check if this email is signed up for the CFC Slack workspace.
+   * 
+   * @param {string} volunteerEmail 
+   */
+  const slackExists = (volunteerEmail) => {
+    fetch(`http://localhost:5000/api/volunteer/slack/exists`, {
+      method: 'POST',
+      body: JSON.stringify({ 
+        email: volunteerEmail 
+      }),
+      headers: {
+        'Content-Type': 'application/json',
+      }
+    })
+    .then((data) => {
+      if (data.status === 404) {
+        console.log(data);
+        throw new Error('404: Route not found.');
+      } else return data.json();
+    })
+    .then((response) => {
+
+      if (response.exists) {  // User found.
+        const slackProfile = {
+          isAuthenticated: true,
+          email: volunteerEmail,
+          notRegistered: false, 
+          suid: response.suid,
+          name: response.name,
+          img:  response.img,
+        };
+        
+        setProfile(slackProfile);
+        // Use localstorage for the moment until we get Sessions figured out.
+        localStorage.setItem('volunteer', JSON.stringify(slackProfile));
+      } else {
+
+        const updatedProfile = {
+          isAuthenticated: false,
+          email: volunteerEmail,
+          notRegistered: true, 
+          registrationStep: 0,
+        };
+        setProfile({});
```

Comment:
Do you need to set the profile to an empty object first? Can't you just set it directly to the new profile object?

Reply:
For some reason it wasn't triggering a re-rendering of the Register page and moving the user to the next step. I meant to go back after everything was refactored and see if it's still necessary. I'll check today.